File: samples/sample.swift
import Foundation

class FooBar {
    var foo: String
    var bar: Int

    init(foo: String, bar: Int) {
        self.foo = foo
        self.bar = bar
    }

    func print() {
        print("Foo: \(self.foo), Bar: \(self.bar)")
    }
}

// Using the class
let foobar = FooBar(foo: "Lorem Ipsum", bar: 123)
foobar.print()

// Array
let baz = [1, 2, 3, 4, 5]
for i in baz {
    print(i)
}

// Dictionary
let qux = ["foo": "bar", "baz": "qux"]
for (key, value) in qux {
    print("Key: \(key), Value: \(value)")
}

// Conditional
if qux.keys.contains("foo") {
    print("Found foo: \(qux["foo"] ?? "")")
} else {
    print("Did not find foo")
}

// Switch statement
switch foobar.bar {
case 123:
    print("Bar is 123")
default:
    print("Bar is not 123")
}

// Function with closure
let numbers = [1, 2, 3, 4, 5]
let doubledNumbers = numbers.map { $0 * 2 }
print(doubledNumbers)

File: samples/sample.scala
import scala.collection.mutable

// Class
class FooBar(val foo: String, val bar: Int) {
  def printFooBar() = println(s"Foo: $foo, Bar: $bar")
}

// Using the class
val foobar = new FooBar("Lorem Ipsum", 123)
foobar.printFooBar()

// Array
val baz = Array(1, 2, 3, 4, 5)
for (i <- baz) println(i)

// Map
val qux = mutable.Map("foo" -> "bar", "baz" -> "qux")
for ((key, value) <- qux) println(s"Key: $key, Value: $value")

// Conditional
if (qux.contains("foo")) println(s"Found foo: ${qux("foo")}")
else println("Did not find foo")

// Match statement
foobar.bar match {
  case 123 => println("Bar is 123")
  case _ => println("Bar is not 123")
}

// Function
def double(x: Int): Int = x * 2
val result = double(6)
println(s"Double of 6 is $result")
